### DISCUSSION TITLE
asset() vs mix() and relative vs absolute URLs

### DISCUSSION THREAD

**Discussion by ethanclevenger91** [laravel/framework#33969] [General]
- Laravel Version: 7.25.0
- PHP Version: 7.4.8
- Database Driver & Version: MySQL (MariaDB), Ver 14.14 Distrib 5.7.29, for Linux (x86_64)

### Description:
The `mix()` helper produces a relative URL unless you've defined `mix_url` in `config/app.php`. Or that's [the implication of this line](https://github.com/laravel/framework/blob/7.x/src/Illuminate/Foundation/Mix.php#L66).

Out of the box, however, `asset()` produces an absolute URL [based on the app's configured URL](https://github.com/laravel/framework/blob/7.x/src/Illuminate/Foundation/helpers.php#L146). 

And in `config/app.php`, we actually have an `asset_url` value out of the box for parity with `mix_url` - `asset()` just doesn't rely on it.

This can result in some edge cases [like this one](https://github.com/barryvdh/laravel-snappy/issues/352) where you end up doing `asset(mix())` to get the absolute URL you want.

Seems like the behavior should be consistent. `asset` should either depend on the config file, or `mix` shouldn't, and having a consistent configuration from one to the other should result in a consistent URL treatment.

Also could be worth considering dropping the `mix_url` config entirely (I think it's undocumented?) and using the same config value to generate both, but that may have side-effects.

### Steps To Reproduce:

1. Create Laravel install with Mix
2. Generate some CSS/JS and insert it in a view using `{{ asset('css/app.css') }}`.
3. See absolute URL in page source
4. Replace with `{{ mix('css/app.css') }}`
5. See relative URL in page source
6. Replace with `{{ asset(mix('css/app.css')) }}`
7. See absolute URL in page source

**Comment by taylorotwell** [comment]
mix_url is used by Laravel Vapor.

**Comment by taylorotwell** [comment]
Since this seems to be more of a feature request than a true "bug", will convert this to a discussion for now.
